test/e2e/atlas/streams_with_clusters_test.go

```diff
@@ -105,6 +106,166 @@ func TestStreamsWithClusters(t *testing.T) {
 		a.Equal(connectionName, *connection.Name)
 		a.Equal("atlasAdmin", *connection.DbRoleToExecute.Role)
 	})
+
+	processorName := "ExampleSP"
+	t.Run("Create streams connection to sample stream", func(t *testing.T) {
+		streamsCmd := exec.Command(cliPath,
+			"streams",
+			"connection",
+			"create",
+			"-f",
+			"data/create_streams_connection_sample.json",
+			"-i",
+			instanceName,
+			"-o=json",
+			"--projectId",
+			g.projectID,
+		)
+
+		streamsCmd.Env = os.Environ()
+		streamsResp, streamsErr := e2e.RunAndGetStdOut(streamsCmd)
+		req.NoError(streamsErr, string(streamsResp))
+
+		var connection atlasv2.StreamsConnection
+		req.NoError(json.Unmarshal(streamsResp, &connection))
+
+		// Assert on config from create_streams_connection_atlas_test.json
+		a := assert.New(t)
+		a.Equal("sample_stream_solar", *connection.Name)
+	})
+
+	t.Run("Create a stream processor with an atlas cluster sink", func(t *testing.T) {
+		streamsCmd := exec.Command(cliPath,
+			"streams",
+			"processor",
+			"create",
+			processorName,
+			"-f",
+			"data/create_stream_processor_test.json",
+			"-i",
+			instanceName,
+			"-o=json",
+			"--projectId",
+			g.projectID,
+		)
+
+		streamsCmd.Env = os.Environ()
+		streamsResp, streamsErr := e2e.RunAndGetStdOut(streamsCmd)
+		req.NoError(streamsErr, string(streamsResp))
+
+		var processor atlasv2.StreamsProcessor
+		req.NoError(json.Unmarshal(streamsResp, &processor))
+
+		a := assert.New(t)
+		a.Equal(processorName, *processor.Name)
+	})
+
+	t.Run("Describe a stream processor with an atlas cluster sink", func(t *testing.T) {
```

Comment:
All of the tests below this one concern me because they are relying on data populated/created by previous t.Run() tests. Each test should be able to function in isolation. If you want to rely on previous configuration, then it should be nested within the test.

I would recommend each of the commands `describe`, `list`, `delete`, `start`, `stop` do all of the creation/setup necessary and just don't run assertions on those steps (since the assertions are covered here). 

An alternative idea is to create a larger E2E test that tests everything. It could create, do assertions on the creation; describe, do assertions on the describe; list and assert; start and assert; stop and assert; and delete and assert.